src/main/java/org/opensearch/flowframework/transport/GetWorkflowStepTransportAction.java

```diff
@@ -47,11 +49,21 @@ public GetWorkflowStepTransportAction(
     }
 
     @Override
-    protected void doExecute(Task task, ActionRequest request, ActionListener<GetWorkflowStepResponse> listener) {
+    protected void doExecute(Task task, WorkflowRequest request, ActionListener<GetWorkflowStepResponse> listener) {
         try {
-            WorkflowValidator workflowValidator = this.workflowStepFactory.getWorkflowValidator();
+            List<String> steps = new ArrayList<>(request.getParams().keySet());
+            WorkflowValidator workflowValidator;
+            if (steps.isEmpty()) {
+                workflowValidator = this.workflowStepFactory.getWorkflowValidator();
+            } else {
+                workflowValidator = this.workflowStepFactory.getWorkflowValidatorByStep(steps);
+            }
             listener.onResponse(new GetWorkflowStepResponse(workflowValidator));
         } catch (Exception e) {
+            if (e instanceof FlowFrameworkException) {
+                logger.error("Please only use only valid step name");
```

Comment:
Users won't see this log message. Probably better to just say "invalid step name" here.  Or just log it inside the validator when you throw the exception, or use the exception message here.